scripts/lib/CIME/BuildTools/macrowriterbase.py

```diff
@@ -15,6 +15,59 @@
 from CIME.XML.standard_module_setup import *
 logger = logging.getLogger(__name__)
 
+def _get_components(value):
+    """
+    >>> value = '-something ${shell ${NETCDF_PATH}/bin/nf-config --flibs} -lblas -llapack'
```

Comment:
This was a big change that I can't quite remember why it was needed. shell and environment variables weren't being handled correctly.